lib/modules/pipeline/index.js

```diff
@@ -82,7 +83,26 @@ class Pipeline {
           });
         });
       },
-      function runWebpack(next) {
+      function shouldRunWebpack(next){
+        // assuming we got here because an asset was changed, let's check our webpack config
+        // to see if the changed asset requires webpack to run
+        if(modifiedAsset){
+          const configReader = new WebpackConfigReader({webpackConfigName: self.webpackConfigName});
+          return configReader.readConfig((err, config) => {
```

Comment:
I think this could be done on init, since the webpackConfig shouldn't change from one build to the other

Reply:
It could change if it's been ejected and the DApp developer modifies it. Alternatively, rather than reloading it, we could have the watcher for `webpack.config.js` warn that embark needs to be restarted if that file changes. I think, though, that restarting the pipeline when it changes and reading it each time make for better UX.

Reply:
I agree that if the user modifies an ejected webpack.config.js, the reading of the config will need to remain in the pipeline, and for this case, we shouldn't move it to init.